content/en/docs/event-sources/kernel/architecture.md

```diff
@@ -0,0 +1,68 @@
+* The Schema version refers to the type of events that the specific driver supports. The [Syscall Events](/docs/reference/rules/supported-events/) documentation page shows the list of fields that are supported for each version of Falco. Every time that list changes the version number is updated as well.
+
+![](/docs/images/kernel_source_start_capture.png)
+
+When running Falco it is possible to verify the currently compatible version numbers with `falco --version`. For instance, this is the output for Falco 0.35.1:
+
+```
+# falco --version
+2023-07-01T16:23:43+0000: Falco version: 0.35.1 (x86_64)
+2023-07-01T16:23:43+0000: Falco initialized with configuration file: /etc/falco/falco.yaml
+Falco version: 0.35.1
+Libs version:  0.11.3
+Plugin API:    3.0.0
+Engine:        17
+Driver:
+  API version:    4.0.0
+  Schema version: 2.0.0
+  Default driver: 5.0.1+driver
+```
+
+Once Falco is running, a stream of events is returned directly from the kernel. libscap's API allow the data to flow with a consistent format from the kernel to userspace.
+
+The main interface that governs this is the scap event format. Once the chosen driver is loaded and initialized in the kernel, the events are encoded with a specific header and a payload:
+
+```Cpp=
+struct ppm_evt_hdr {
+	uint64_t ts; /* timestamp, in nanoseconds from epoch */
+	uint64_t tid; /* the tid of the thread that generated this event */
+	uint32_t len; /* the event len, including the header */
+	uint16_t type; /* the event type */
+	uint32_t nparams; /* the number of parameters of the event */
+};
+```
+
+The payload contains an array of lengths of each parameter followed by the content of the parameters themselves. The parameter type is a numeric identifer that maps with each event documented in the reference.
+
+For example, the `dup3` event is defined in the reference as:
+
+**dup3**(FD res, FD oldfd, FD newfd, FLAGS32 flags: *O_LARGEFILE*, *O_DIRECTORY*, *O_DIRECT*, *O_TRUNC*, *O_SYNC*, *O_NONBLOCK*, *O_EXCL*, *O_DSYNC*, *O_APPEND*, *O_CREAT*, *O_RDWR*, *O_WRONLY*, *O_RDONLY*, *O_CLOEXEC*, *O_NONE*, *O_TMPFILE*) 
+
+Meaning that its encoding will be composed of an header containing the timestamp and tid, `nparams` will be 4 and the complete encoding will be:
+
+```
+[header] [uint16(8)] [uint16(8)] [uint16(8)] [uint16(32)] [res] [oldfd] [newfd] [flags]
+```
+
+![](/docs/images/kernel_source_capture.png)
+
+All the supported events can be used in Falco rules, can be enabled or disabled if necessary and are documented in the reference.
```

Comment:
```suggestion
All the supported events can be used in Falco rules, can be enabled or disabled if necessary and are documented in the reference.
```

I don't believe this is true in all cases. There can be internal events that aren't available in the rules.